lib/iris/coords.py

```diff
@@ -1338,17 +1423,18 @@ class DimCoord(Coord):
     @staticmethod
     def from_coord(coord):
         """Create a new DimCoord from the given coordinate."""
-        return DimCoord(coord.points, standard_name=coord.standard_name,
+        return DimCoord(coord.core_points(), standard_name=coord.standard_name,
                         long_name=coord.long_name, var_name=coord.var_name,
-                        units=coord.units, bounds=coord.bounds,
+                        units=coord.units, bounds=coord.core_bounds(),
                         attributes=coord.attributes,
                         coord_system=copy.deepcopy(coord.coord_system),
                         circular=getattr(coord, 'circular', False))
 
-    @classmethod
-    def from_regular(cls, zeroth, step, count, standard_name=None,
+    @staticmethod
+    def from_regular(zeroth, step, count, standard_name=None,
```

Comment:
By contrast, I *agree* with going through the init so we don't bypass checks.
However, I think this should be a classmethod, and create the result by calling `cls`, as that makes it work for possible inheriting subclasses.